internal/fs/inode/file.go

```diff
@@ -667,6 +722,11 @@ func (f *FileInode) Sync(ctx context.Context) (err error) {
 	}
 
 	// If we wrote out a new object, we need to update our state.
+	f.updateInodeStateAfterSync(newObj)
+	return
+}
+
+func (f *FileInode) updateInodeStateAfterSync(newObj *gcs.Object) {
```

Comment:
Sorry i missed this earlier, to avoid more dependencies on getting gcs.Object from bucket_handle whereas we use only gcs.MinObject everywhere, can you return gcs.MinObject directly for finalizeUpload method.

Reply:
Done in separate PR: https://github.com/GoogleCloudPlatform/gcsfuse/pull/2795